### ISSUE TITLE
AttributeError: attribute 'lemma_' of 'spacy.tokens.token.Token' objects is not writable

### ISSUE THREAD

**Issue by schlichtanders** [explosion/spaCy#973]
I tried to implement a simple lemmatization for German via a custom addition to the nlp pipeline:
```python
from textblob_de import WordList
import spacy
nlp = spacy.load("de")

def textblob_lemmas(doc):
    for sent in doc.sents:
        lemmas = WordList([str(w) for w in sent]).lemmatize()
        for w, l in zip(sent, lemmas):
            w.lemma_ = l

nlp.pipeline = [nlp.tagger, nlp.parser, nlp.entity, textblob_lemmas]
nlp("Dies ist ein Test.")
```
However unfortunately, this raises ``AttributeError: attribute 'lemma_' of 'spacy.tokens.token.Token' objects is not writable``.

[Same happens if I use 
```
def custom_pipeline(nlp):
    return (nlp.tagger, nlp.parser, textblob_lemmas, nlp.entity)

nlp = spacy.load('de', create_pipeline=custom_pipeline)
```
for defining ``nlp`` instead]

## Your Environment

* Operating System: Windows 10, 64bit
* Python Version Used: 3.6
* spaCy Version Used: 1.7.5
* Environment Information:

**Comment by schlichtanders** [comment]
It looks like the only thing missing are the respective setter methods which already exist for ``tag_``, ``ent_id_`` and others.
lemma: https://github.com/explosion/spaCy/blob/master/spacy/tokens/token.pyx#L191-L193
lemma_: https://github.com/explosion/spaCy/blob/master/spacy/tokens/token.pyx#L551-L553

why not to add these setters to other (or maybe all) basic attributes?

**Comment by schlichtanders** [comment]
It would really really helpful, if this could be changed rather soon, as it enables way better use of spacy in total. It seems to be only a couple of lines of code involved.

Currently, in our usecase we combine spacy with extra lemmas for instance, because there seems to be no way to include them into spacy itself. It would be awesome to just pass around spacy documents/spans/tokens - I think this is also one of the original intentions of spacy, isn't it?

**Comment by honnibal** [comment]
Agreed, this would be better. Do you mind making the pull request?

**Comment by schlichtanders** [comment]
Yeah, I like to do it - will be after eastern.
However i fear that maybe problems regarding serialization​ may occur...
Do you know points I should be aware of?

On Apr 13, 2017 12:25 PM, "Matthew Honnibal" <notifications@github.com>
wrote:

Agreed, this would be better. Do you mind making the pull request?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/explosion/spaCy/issues/973#issuecomment-293851829>, or mute
the thread

**Comment by ines** [comment]
Fixed in v1.8.0!

**Comment by schlichtanders** [comment]
@ines thanks a lot for fixing this!
(I apparently had not enough time to look into this again, the more thank you that this was not forgotten)
